Core/src/main.cpp

```diff
@@ -59,56 +59,52 @@ int run_glfw() {
     window.SetWindowToCurrentThread();
     window.SetKeyCallback(Input::KeyCallback);
     window.SetMouseButtonCallback(Input::MouseButtonCallback);    
-	
-	// Load GLAD
-	if (!gladLoadGLLoader((GLADloadproc)glfwGetProcAddress))
-	{
-		std::cout << "Failed to initialize GLAD" << std::endl;
-		return -1;
-	}
-
-	// Set size of framebuffer
+    
+    // Load GLAD
+    if (!gladLoadGLLoader((GLADloadproc)glfwGetProcAddress))
+    {
+        std::cout << "Failed to initialize GLAD" << std::endl;
+        return -1;
+    }
+
+    // Set size of framebuffer
     glViewport(0, 0, window.GetWidth(), window.GetHeight());
     window.SetFramebufferSizeCallback();
 
-	// ECS ======================================================
+    // ECS ======================================================
     std::vector<System*> systems;
+
```

Comment:
It doesn't need to, but its more that why is the UI dependent on the scene being created. The initialize function for audio creates the fmod system with a channel count. If the scene is instantiated first, you attempt to add audio functionality without a system to add to. It makes sense that the scene should be dependent on the systems being instantiated, rather than the systems being depended on the scene. Please respond

Reply:
See top level review comment

```
RE: Comments in AudioComponent, AudioListener and AudioManager and its relationship to Scene -
In summary you should probably split AudioManager (which is actually a system) into an actual FModManager.h class (for which a singleton would be perfectly acceptable and do all the variable associations in your Listeners and audio components) and an Audio.h, Audio.cpp ECS SYSTEM. Which behaves like the rest of the systems without interdependencies.

FModManager::getInstance()->GetFMODSystem()-><do something>

Now we dont have to store a system as a variable, we dont have to pass it in as a param, we dont have to make order dependent code
```

Reply:
I disagree with your last comment. 


Systems act and process entities. Entities just contain data. 
Entities are set up in the scene definition to be acted upon. 

an ECS Audio system reaches into the scene definition - which is just a collection of inert collections of data - and then acts upon them.

Reply:
You're right that UI shouldn't be dependent on the scene being created, but no system should be dependent on it anyways. Likewise, no component should be dependent on a system existing either. In UI's case this is an easy refactor with ```Core::getInstance()->GetScene()```, but here we don't have something similar.

Like Richard says, Systems and Components/Entities should no be dependent on either one existing.
Components are data. Systems act on that data.
If there are no components for a System to act upon, then it does nothing.
If there's no System to act on a component, then the component just exists.

Going back to top level review, I'm in agreement with Richard. What is Audio Manager supposed to be, a System or a Manager? If audio components need to have an instance of FMOD to be able to load audio, then we can have an instance of FMOD in an AudioManager. BUT then we need to separate functionality and have a different AudioSystem alongside it.